packages/wdio-cli/src/commands/config.ts

```diff
@@ -65,7 +65,7 @@ const runConfig = async function (useYarn: boolean, yes: boolean, exit = false)
         if (!hasPackage('ts-node')) {
             packagesToInstall.push('ts-node', 'typescript')
         }
-        if (!hasFile('tsconfig.json')){
+        if (answers.generateTSConfigFile) {
```

Comment:
So I suggest to remove this if statement and always generate a tsconfig file.

Reply:
@christian-bromann 

what if there already exists a tsconfig file as the main issue mentioned. we need that check to know whether to create the tsconfig in current directory or inside test

Reply:
Why don't we just always create a custom tsconfig file in `./test`? I think it makes the codebase easier and for the user it doesn't really matter, right?

Reply:
@christian-bromann  As mentioned before, if we create inside test , then wdio.config should be also inside test folder . So the structure ts project looks different than for all other compilers . 

in our previous comment it was decided that we will create inside test folder only if a tsconfig file exists in the root directory

Reply:
https://github.com/webdriverio/webdriverio/issues/7640#issuecomment-972694037

Reply:
> then wdio.config should be also inside test folder . So the structure ts project looks different than for all other compilers 

We can just set [`autoCompileOpts `](https://webdriver.io/docs/typescript#configuration) to point to the correct `tsconfig.json` as you already implemented it in this PR.

> in our previous comment it was decided that we will create inside test folder only if a tsconfig file exists in the root directory

Correct.

Reply:
@christian-bromann 

> in our previous comment it was decided that we will create inside test folder only if a tsconfig file exists in the root directory - correct

This is how its implemented now with autocompile option be added if tsconfig file is not in root . So i will keep it the way it is ? or you want tsconfig to be in test folder in all cases ?

Reply:
> or you want tsconfig to be in test folder in all cases ?

Let's always put it in `/test` because it simplifies the implementation and it makes no difference for the user.